mdc_weight_mgmt/models/mdc_weight_record.py

```diff
@@ -143,29 +153,19 @@ def write(self, vals):
                 raise ValidationError(_("The end date must be later than the start date."))
         return res
 
-    @api.constrains('equipment_id', 'start', 'end')
-    def _check_overlap_constrains(self):
-        for record in self:
-            if not record.equipment_id or not record.start or not record.end or self.env.context.get('mdc_cron_create', False):
-                continue
-            msg = self._check_overlap_manual(record.equipment_id, record.start, record.end)
-            if msg:
-                raise ValidationError(msg)
-
-    @api.model
     def _check_overlap_manual(self, equipment_id, start, end):
         overlapping = self.search([
-            ('equipment_id', '=', equipment_id.id),
+            ('equipment_id', '=', equipment_id),
```

Comment:
I'll give you an example for `equipment_id` possible dual value (between record and int value)